source/appModules/foobar2000.py

```diff
@@ -34,23 +34,44 @@ def getElapsedAndTotal(self):
 			ltime = text.split("|")[4].split(" / ")
 		except IndexError:
 			return None
-		elapsedTime = calendar.timegm(time.strptime(ltime[0].strip(),getFormat(ltime[0])))
-		totalTime = calendar.timegm(time.strptime(ltime[1].strip(),getFormat(ltime[1])))
-		return elapsedTime,totalTime
+		return ltime
 
-	def script_reportRemainingTime(self,gesture):
-		times=self.getElapsedAndTotal()
-		if times is None:
+	def getElapsedAndTotalIfPlaying(self):
+		times = self.getElapsedAndTotal()
+		if not times:
 			# Translators: Reported when no track is playing in Foobar 2000.
 			ui.message(_("No track playing"))
-			return
-		elapsedTime,totalTime = times
-		remainingTime = totalTime-elapsedTime
-		msg = time.strftime(getOutputFormat(remainingTime),time.gmtime(remainingTime))
-		ui.message(msg)
-	# Translators: The description of an NVDA command for Foobar 2000.
+		return times
+
+	def script_reportRemainingTime(self,gesture):
+		times = self.getElapsedAndTotalIfPlaying()
+		if times is not None:
+			elapsedTime = calendar.timegm(time.strptime(times[0].strip(),getFormat(times[0])))
+			totalTime = calendar.timegm(time.strptime(times[1].strip(),getFormat(times[1])))
+			remainingTime = totalTime-elapsedTime
+			msg = time.strftime(getOutputFormat(remainingTime),time.gmtime(remainingTime))
+			ui.message(msg)
+	# Translators: The description of an NVDA command for reading the remaining time of the currently playing track in Foobar 2000.
 	script_reportRemainingTime.__doc__ = _("Reports the remaining time of the currently playing track, if any")
 
+	def script_reportElapsedTime(self,gesture):
+		times = self.getElapsedAndTotalIfPlaying()
+		if times is not None:
+			elapsedTime,totalTime = times

```

Comment:
is it really valid to do this? I suspect this is incorrect and you want something like:

```
elapsedTime = calendar.timegm(time.strptime(times[0].strip(),getFormat(times[0])))
totalTime = calendar.timegm(time.strptime(times[1].strip(),getFormat(times[1])))
```

If so perhaps put that code into a helper function so that it is not duplicated.